src/goalServices/goals.js

```diff
@@ -1277,7 +1305,34 @@ export async function saveGoalsForReport(goals, report) {
     },
   });
 
-  const currentGoals = await Promise.all(goals.map(async (goal) => {
+  // Loop and Create or Update goals.
+  const currentGoals = await Promise.all(goals.map(async (goal, index) => {
+    // We need to skip creation of monitoring goals for non monitoring grants.
+    if (goal.standard === 'Monitoring') {
+      // Find the corresponding monitoring goals.
+      const monitoringGoals = await Goal.findAll({
+        attribute: ['grantId'],
+        raw: true,
+        where: {
+          grantId: goal.grantIds,
+          standard: 'Monitoring',
```

Comment:
It looks like the standard column is not on the Goal

Reply:
Updated to be createdVia = 'monitoring'.

Reply:
seems like this might benefit from some more unit testing if it was passing here and with my comment on line 1325